### PR TITLE
Block assertion regeneration after audit starts

### PR DESCRIPTION
**Pull request by vteague**
Along with tests that the two allowed ASM states work and the others throw an exception.

### CODE REVIEW

**Comment by vteague** [comment]
I can't actually figure out how to test this for the presence of ComparisonAudits, because (quite rightly) it's hard to get it into the state where it has ComparisonAudits but is not in one of the states where assertion generation blocks anyway.

File: server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  /**
+   * Container for the mock-up database.
+   */
+  private final static PostgreSQLContainer<?> postgres = createTestContainer();
+
+  /**
+   * The Generate Assertions endpoint.
+   */
+  private final GenerateAssertions endpoint = new GenerateAssertions();
+
```
Comment (michelleblom) [inline server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java:87]: Might be worth, for anyone reading this, to have some more detail about the parameters being supplied to the GenerateAssertionsResponse/Request parameters. eg. "Mock response for tinyExample1 contest where X is true and Y is false" or something like that. And similarly for the request, "Request for tinyExample1 contest with an allowed time of 5 seconds."

File: server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+   */
+  private final static GenerateAssertionsRequest tinyIRVRequest
+      = new GenerateAssertionsRequest(tinyIRV, tinyIRVCount, 5,
+      tinyIRVCandidates.stream().map(Choice::name).toList());
+
+  /**
+   * Raire endpoint for getting assertions.
+   */
+  private final String raireGenerateAssertionsEndpoint = "/raire/generate-assertions";
+
```
Comment (michelleblom) [inline server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java:105]: Can the port be specified somewhere else in settings ... what if 8110 clashes with something else someone is running?

**Review by michelleblom — APPROVED** [review]

File: server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java (hunk 2 above)
Comment (vteague) [reply server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/endpoint/GenerateAssertionsAPITests.java:105]: Excellent idea; done in the other PR.

**Pull request merged**